src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1285,6 +1298,161 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
+                This parameter does not filter out data. It must be provided when `column` is specified.
+            num_partitions: number of partitions to create when reading in parallel from multiple processes and threads.
+                It must be provided when `column` is specified.
+            query_timeout: The timeout (in seconds) for each query execution. A default value of `0` means
+                the query will never time out. The timeout behavior can also be configured within
+                the `create_connection` method when establishing the database connection, depending on the capabilities
+                of the DBMS and its driver.
+            fetch_size: The number of rows to fetch per batch from the external data source.
+                This determines how many rows are retrieved in each round trip,
+                which can improve performance for drivers with a low default fetch size.
+            custom_schema: a custom snowflake table schema to read data from external data source,
+                the column names should be identical to corresponded column names external data source.
+                This can be a schema string, for example: "id INTEGER, int_col INTEGER, text_col STRING",
+                or StructType, for example: StructType([StructField("ID", IntegerType(), False), StructField("INT_COL", IntegerType(), False), StructField("TEXT_COL", StringType(), False)])
+            predicates: A list of expressions suitable for inclusion in WHERE clauses, where each expression defines a partition.
+                Partitions will be retrieved in parallel.
+                If both `column` and `predicates` are specified, `column` takes precedence.
+            session_init_statement: One or more SQL statements executed before fetching data from
+                the external data source.
+                This can be used for session initialization tasks such as setting configurations.
+                For example, `"SET TRANSACTION ISOLATION LEVEL READ UNCOMMITTED"` can be used in SQL Server
+                to avoid row locks and improve read performance.
+                The `session_init_statement` is executed only once at the beginning of each partition read.
+        """
+        if (not table and not query) or (table and query):
+            raise SnowparkDataframeReaderException(
+                "Either 'table' or 'query' must be provided, but not both."
+            )
+        external_access_integration = udtf_configs.get(
+            "external_access_integration", None
+        )
+        imports = udtf_configs.get("imports", None)
+        packages = udtf_configs.get("packages", ["com.snowflake:snowpark:latest"])
+        java_version = udtf_configs.get("java_version", 11)
+        eai_info = self._session.sql(
+            f"describe external access integration {external_access_integration}"
+        ).collect()
+        secret = None
+        for r in eai_info:
```

Comment:
1. secret = None, raise error
2. len(secret) = 1, we just use this one, 
3. len(secret) > 1, we need extra user info, we require users to give us the secret name

----

case 2:
solution 1: secret name is required, we validate

Reply:
check with Qinyi

Reply:
Aligned with Qinyi: Secret will be a required field in udtf_configs